PubHub/PubHub.API/Domain/Entities/BookAuthor.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace PubHub.API.Domain.Entities
+{
+    public class BookAuthor
+    {
+        public int BookId { get; set; }
+        public int AuthorId { get; set; }
+
+        #region Navs
+        public Book? Book { get; set; }
+        public Author? Author { get; set; }
```

Comment:
`public Book Book { get; set; } = null!` instead? - and the same for author?
I think it would be easier to work with, code wise, when writing LINQ queries (no need to check for null or insert `!`). Additionally, I think it better clarifies that, even though we don't necessarily want to carry around an object instance all the time, there is supposed to be a non-nullable relation to another object.

Reply:
I don't agree.

`Book` can be `null`, as we're eager loading our relations. This can then result in `NullReferenceExceptions` being thrown if we forget to specifically load the entity.
Every nav prop should, in my opinion be a nullable type, except from collections, as they should never be `null`.

Reply:
But how are we gonna do LINQ queries without using `!` then? We can't use `?`:
![image](https://github.com/Mike-Mortensen-Portfolio/PubHub_H6_Final/assets/122673993/eeb5a45e-693b-4a75-bf98-6f95f88a4236)
Are we just gonna do like so?:
![image](https://github.com/Mike-Mortensen-Portfolio/PubHub_H6_Final/assets/122673993/c907010b-b8a0-4881-a2a0-eac8e0a485ca)

Reply:
You're not suppoed to.

Keep in mind that when you explicitly include `Account` you can vouch for its precense, hence you can do `p.Account!.Email` to signify that you believe that it's not null here, even though the Nullable system can't track it.

Reply:
Yeah, that does make good sense and clearly show the intent right where it happens. I will resolve the other comments 👍